proxy/http/HttpTransact.cc

```diff
@@ -6622,7 +6622,8 @@ HttpTransact::handle_content_length_header(State *s, HTTPHdr *header, HTTPHdr *b
     TxnDebug("http_trans", "[handle_content_length_header] RESPONSE cont len in hdr is %" PRId64, header->get_content_length());
   } else {
     // No content length header.
-    if (s->source == SOURCE_CACHE) {
+    if (s->source == SOURCE_CACHE ||
+        (s->source == SOURCE_HTTP_ORIGIN_SERVER && s->hdr_info.server_response.status_get() == HTTP_STATUS_NOT_MODIFIED)) {
       // If there is no content-length header, we can
```

Comment:
Call to object_size_get() in this block crashes in the http_with_origin test

Reply:
I have modified the code to check if the object exists in cache as http_with_origin_test responds with 304 for unconditional request.